### PR TITLE
feat: tesseract implementation

### LINKED ISSUE #23: Implement tesseract video-to-text
# Issue description
Implement tesseract js methods for live text recognition in the video feed provided by #17 
This should already take into account the different rectangles (capture areas) that define where text should be scanned.

See https://maheshlocker.com/building-a-real-time-video-text-recognition-app-using-tesseract-js-and-react for reference.

Tesseract examples with rectangles & multiple workers: https://github.com/naptha/tesseract.js/blob/master/docs/examples.md

- [ ] scale capture area width to window(mediastream) width

# Definition Of Ready (DoR)
*This issue can be worked on if*
- [x] a developer is assigned
- [x] the issue is labeled
- [x] the issue is assigned to the current sprint
- [x] the issue is estimated
- [x] a meaningful title has been set
- [x] the issue is described in detail

### PR DESCRIPTION
**Pull request by Claiyc**
# PR description
*Describe your changes in detail here*
Once merged, this will add image-to-text translation using tesseract.js (not using regex recognition yet).
This will take capture areas into consideration (only extract text in capture areas)

As of now this will just print the recognized text into the run-view.

# Definition Of Done (DoD)
*This PR can be squashed / merged if*
- [x] a developer is assigned
- [x] the PR is NOT estimated
- [x] the PR is labeled
- [x] the PR is assigned to the current sprint
- [x] a meaningful title has been set according to https://www.conventionalcommits.org/
- [x] the PR is described in detail
- [x] the PR links to an issue
- [x] the PR has been reviewed

*Add additional conditions here if necessary for this PR*
fixes #23

### CODE REVIEW

**Comment by Claiyc** [comment]
First part of implementation. Currently works as "call once, run once" -> should of course work as background-loop in the end. Mapping of capture areas to actual stream is not correct yet.
As of now it's quite the ugly chaining of async functions.. Will probably have to iterate on that.

File: index.html [hunk 1]
```diff
@@ -4,10 +4,7 @@
         <meta charset="UTF-8" />
         <link rel="icon" href="/favicon.ico" />
         <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-        <meta
-            http-equiv="Content-Security-Policy"
-            content="script-src 'self' 'unsafe-inline';"
-        />
+        <!-- TODO Add Content Security Policy-->
```
Comment (BeierKevin) [inline index.html:7]: Created a bug issue for it. #62

File: src/proc/Vigad.ts [hunk 2]
```diff
@@ -76,4 +84,20 @@ export class Vigad {
...
+            this.intervalRunning = true;
+            console.log("started tesseract");
+        }
+    }
+
+    public stopTesseract(): void {
+        clearInterval(this.tesseractInterval);
+        this.intervalRunning = false;
+        console.log("stopped tesseract");
+    }
```
Comment (BeierKevin) [inline src/proc/Vigad.ts:102]: Before merging, remove console.log

File: src/proc/TesseractHandler.ts [hunk 3]
```diff
@@ -1,8 +1,102 @@
...
+     */
+    public static getInstance(): TesseractHandler {
+        if (!this.instance) {
+            this.instance = new this();
+        }
+
+        return this.instance;
+    }
+
+    public isRunning(): boolean {
```
Comment (BeierKevin) [inline src/proc/TesseractHandler.ts:32]: Proper documentation for these functions before merge, and before this PR isn't a draft anymore

**Review by BeierKevin — COMMENTED** [review]
Over all, the implementation looks quite good, but we will see how good this works together with the regex matcher.

File: src/proc/Vigad.ts (hunk 2 above)
Comment (Claiyc) [reply src/proc/Vigad.ts:102]: I actually kept them on purpose as I think it's something that can be quite useful for debugging. Of course a real logger would be better but this doesn't hurt, does it?

**Review by BeierKevin — APPROVED** [review]
Looks good to me.

**Pull request merged**